### ISSUE TITLE
Crash (segfault) in monster::plan() at src/monmove.cpp:219

### ISSUE THREAD

**Issue by dholmes215** [CleverRaven/Cataclysm-DDA#11025]
I was duking it out with some zombies when this happened.

Version was approximately f9ee723.

I still have it in the debugger if anybody wants me to investigate further.

```
#0  0x0000000100aa7196 in __gnu_cxx::new_allocator<std::_Rb_tree_node<int> >::construct<int, int const&> (this=0x229cc0, __p=0x6004adc60) at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/ext/new_allocator.h:120
#1  0x0000000100b61456 in std::allocator_traits<std::allocator<std::_Rb_tree_node<int> > >::_S_construct<int, int const&> (__a=..., __p=0x6004adc60)
    at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/bits/alloc_traits.h:253
#2  0x0000000100b61566 in std::allocator_traits<std::allocator<std::_Rb_tree_node<int> > >::construct<int, int const&> (__a=..., __p=0x6004adc60)
    at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/bits/alloc_traits.h:399
#3  0x0000000100c5ca15 in std::_Rb_tree<int, int, std::_Identity<int>, std::less<int>, std::allocator<int> >::_M_create_node<int const&> (this=0x229cc0)
    at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/bits/stl_tree.h:421
#4  0x0000000100c5c828 in std::_Rb_tree<int, int, std::_Identity<int>, std::less<int>, std::allocator<int> >::_M_clone_node (this=0x229cc0, __x=0x200632f7f0)
    at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/bits/stl_tree.h:445
#5  0x0000000100c5ce4b in std::_Rb_tree<int, int, std::_Identity<int>, std::less<int>, std::allocator<int> >::_M_copy (this=0x229cc0, __x=0x200632f7f0, __p=0x60585d0a0)
    at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/bits/stl_tree.h:1219
#6  0x0000000100c5d0fd in std::_Rb_tree<int, int, std::_Identity<int>, std::less<int>, std::allocator<int> >::_Rb_tree (this=0x229cc0, __x=...)
    at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/bits/stl_tree.h:676
#7  0x0000000100b89650 in std::set<int, std::less<int>, std::allocator<int> >::set (this=0x229cc0, __x=...) at /usr/lib/gcc/x86_64-pc-cygwin/4.9.2/include/c++/bits/stl_set.h:197
#8  0x00000001006b0a14 in monster::plan (this=0x6070af950, factions=...) at src/monmove.cpp:219
#9  0x000000010064d0d9 in game::monmove (this=0x6fffff90010) at src/game.cpp:6500
#10 0x0000000100629a69 in game::do_turn (this=0x6fffff90010) at src/game.cpp:1480
#11 0x00000001009ea179 in main (argc=0, argv=0x22ab38) at src/main.cpp:295
```

**Comment by KA101** [comment]
That's the faction-handling stuff by the look of things, just merged it a few days ago.  Sorry.

**Comment by Coolthulhu** [comment]
I think it's the own faction finding code. It shouldn't ever be unable to find own faction, but apparently it does.
I'll make it display a debug message instead of crashing.

**Comment by kevingranade** [comment]
Fixed by #11053
